experimental/adapter-oauth/oauth_adapter.js

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// index.js is used to setup and configure your bot
+
+const { FacebookAdapter } = require('botbuilder-adapter-facebook');
+
+const OAUTH_ENDPOINT = 'https://api.botframework.com';
+const US_GOV_OAUTH_ENDPOINT = 'https://api.botframework.azure.us';
+const botframework_connector_1 = require("botframework-connector");
+const botbuilder_1 = require("botbuilder");
+
+class OAuthAdapter extends FacebookAdapter{
+    constructor({appId: appId, password: password, fb_verify_token: fb_verify_token, fb_password: fb_password, fb_access_token: fb_access_token}){
+        super({
+            verify_token: fb_verify_token,
+            app_secret: fb_password, 
+            access_token: fb_access_token 
+       });
+
+       this.credentials = new botframework_connector_1.MicrosoftAppCredentials(appId, password);
+       this.credentialsProvider = new botframework_connector_1.SimpleCredentialProvider(appId, password);
+    }
+
+    getUserToken(context, connectionName, magicCode) {
+        return __awaiter(this, void 0, void 0, function* () {
```

Comment:
these _awaiter things are the result of this originally being copy pasted from compiled Typescript. We should do this with the normal async syntax and return the __awaiter stuff.

Reply:
Done